lmfdb/hilbert_modular_forms/test_hmf.py

```diff
@@ -158,12 +158,9 @@ def test_download_magma(self):
             page = self.tc.get('/ModularForm/GL2/TotallyReal/{}/holomorphic/{}/download/magma'.format(field, label)).get_data(as_text=True)
             assert expected in page
             assert  'make_newform'  in page
-            try:
-                magma_code = page + '\n';
-                magma_code += 'f, iso := Explode(make_newform());\n'
-                magma_code += 'assert(&and([iso(heckeEigenvalues[P]) eq HeckeEigenvalue(f,P): P in primes[1..10]]));\n'
-                magma_code += 'f;\n'
-                assert 'success' in magma.eval(magma_code)
-            except RuntimeError:
-                pass
```

Comment:
Currently, this `try...except` block makes the test pass even if the magma code is populated with erroneous code which doesn't compile in magma.

Reply:
A small suggestion, as the server where we ran automatic tests don't have magma:
```
magma_code = ....
try:
    assert 'success' in magma.eval(magma_code)
except RuntimError:
    try: 
        assert 'success' in magma_free(magma_code)
    except socket.timeout as err:
            print("Connecting with magma.maths.usyd.edu.au timed out")
            print(err)

Reply:
The problem with RuntimeError when calling `magma.eval(magma_code)` is that *all errors* manifest as RuntimeError; e.g.:

```
sage: magma.eval("assert 1 eq 2;")  # raises RuntimeError, not AssertionError
sage: magma.eval("boaty mcboatface rocks!")  # raises RuntimeError, not SyntaxError
```

However, in my latest commit, I've used your idea of using `magma_free` and catching the timeout, but replacing the outer `try ... except` with `if magma.is_local()`:

```
if magma.is_local():
    assert 'success' in magma.eval(magma_code)
else:
    try:
        from sage.all import magma_free
        assert 'success' in magma_free(magma_code)
    except socket.timeout as err:
        print("Connecting with magma.maths.usyd.edu.au timed out")
        print(err)
```

I don't know how Travis will react to that ... I'll wait and see; but it works locally on legendre

Reply:
OK so Travis didn't like that; apparently `magma.is_local()` was True even though it doesn't have magma. The latest commit tries the following:

```
try:
    assert 'success' in magma.eval(magma_code)
except RuntimeError as the_error:
    if str(the_error).startswith("unable to start magma"):
        try:
            from sage.all import magma_free
            assert 'success' in magma_free(magma_code)
        except socket.timeout as err:
            print("Connecting with magma.maths.usyd.edu.au timed out")
            print(err)
            pass
    else:
        raise
```

Reply:
So that didn't work either; the issue with the above is that `magma_code` is enormous, and waay bigger than the 50,000 bytes limit for the free online magma calculator, so the input is ignored, and returns an empty magma expression, so the assertion fails.

At this point I think all bets are off for executing the magma code on Travis, so I'll aim to check the magma compilation works only when full magma is able to start:

```
try:
    assert 'success' in magma.eval(magma_code)
except RuntimeError as the_error:
    if str(the_error).startswith("unable to start magma"):
        pass
    else:
        raise
```

Reply:
Sounds good!